R/dash.R

```diff
@@ -593,17 +593,47 @@ Dash <- R6::R6Class(
     # ------------------------------------------------------------------------
     callback = function(output, params, func) {
       assert_valid_callbacks(output, params, func)
-
+      
       inputs <- params[vapply(params, function(x) 'input' %in% attr(x, "class"), FUN.VALUE=logical(1))]
       state <- params[vapply(params, function(x) 'state' %in% attr(x, "class"), FUN.VALUE=logical(1))]
-
+      
       if (is.function(func)) {
         clientside_function <- NULL
+      } else if (is.character(func)) {
+        
+        # if namespace not provided, default to clientside        
+        namespace <- "_dashprivate_clientside"
+        
+        fn_name <- paste0("_dashprivate_clientside_", createCallbackId(output))
+        
+        # hashing the name while keeping it a valid JS function name
+        fn_name <- paste0("_", digest(fn_name, 
+                                      "md5", 
+                                      serialize = FALSE))
+        
+        clientside_function <- list(namespace = namespace,
+                                    function_name = fn_name)
+        
+        # register the function with its hashed name representation
+        js_map <- insertIntoJSMap(private$js_map,
+                                  func,
+                                  fn_name)
+        
+        # check that no entries in the inline JS map have names matching those in the
+        # callback map, and vice-versas
+        if (length(intersect(names(private$callback_map), 
+                             sub("_dashprivate_clientside_", "", names(private$js_map)))) > 0) {
+          stop(sprintf("One or more outputs are duplicated across callbacks. Please ensure that all ID and property combinations are unique."), call. = FALSE)
```

Comment:
Isn't this taken care of by `insertIntoCallbackMap` (that this still goes through below)?